app/controllers/passwordless/sessions_controller.rb

```diff
@@ -42,7 +42,7 @@ def show
       BCrypt::Password.create(params[:token])
 
       session = find_session
-      sign_in session.authenticatable
+      sign_in session.authenticatable unless session_expired session
 
       redirect_enabled = Passwordless.redirect_back_after_sign_in
```

Comment:
We don't want to redirect back if the authentication wasn't successful. In that case the user is going to be turned away again and again in an infinite loop.

Instead we would probably want to always redirect to `root_path` and set a `flash[:error]` message about the session being expired.